### PR TITLE
Fix ApplicationThreadPoolScope without disposable

### PR DESCRIPTION
**Pull request by rli**
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by manodnyab — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/ProjectLevelCoroutineScopeTracker.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import com.intellij.openapi.components.ServiceManager
+import com.intellij.openapi.project.Project
+import software.aws.toolkits.jetbrains.utils.ApplicationThreadPoolScope
+import kotlin.reflect.KClass
+
+class ProjectLevelCoroutineScopeTracker(
+    @SuppressWarnings("UnusedProperty")
+    private val project: Project
+) : Disposable {
+    private val scopes: MutableMap<String, ApplicationThreadPoolScope> = mutableMapOf()
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/core/ProjectLevelCoroutineScopeTracker.kt:16]: Should this be a weak hash map?

Are the benefits to caching the scope?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/ProjectLevelCoroutineScopeTracker.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    override fun dispose() { }
+
+    companion object {
+        fun getInstance(project: Project) = ServiceManager.getService(project, ProjectLevelCoroutineScopeTracker::class.java)
+    }
+}
+
+fun Project.applicationThreadPoolScope(coroutineName: String) = ProjectLevelCoroutineScopeTracker.getInstance(this).applicationThreadPoolScope(coroutineName)
+
+fun Project.applicationThreadPoolScope(clazz: KClass<out Any>) = ProjectLevelCoroutineScopeTracker.getInstance(this).applicationThreadPoolScope(clazz.java.name)
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/core/ProjectLevelCoroutineScopeTracker.kt:29]: If we inline this we do not need to pass the class

```
inline fun <reified T : Any> T.applicationThreadPoolScope() = applicationThreadPoolScope(T::class.java.name)
```

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/ProjectLevelCoroutineScopeTracker.kt (hunk 1 above)
Comment (rli) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/core/ProjectLevelCoroutineScopeTracker.kt]: changed to weak value map. re: caching, for our use case i think we can go either way

**Pull request merged**